lib/kan/abilities.rb

```diff
@@ -8,6 +10,8 @@ module ClassMethods
       DEFAULT_ROLE_NAME = :base
       DEFAULT_ROLE_BLOCK = proc { true }
 
+      attr_accessor :logger
```

Comment:
Why getter in class methods? 🤔

Reply:
We have to use instance variable for this instead class methods

Reply:
In this case logger use class scope and methods
```
class Post::Abilities
  include Kan::Abilities

  register 'read' do |user, _|
    logger # => logger instance
    logger.info 'Something happened'
    true
  end
end
```
Need to think how to pass logger from ability instance to action.

Reply:
just use `#instance_exec` :)

```ruby
class Abilities
  def initialize(rules)
    @rules = rules
  end

  def ability
    rule = @rules[:post]
    proc { |payload| instance_exec(payload, &rule) }
  end

  def logger
    'I am logger'
  end
end

rule = proc { |params| puts "logger: #{logger}; hello; #{params}" }
a = Abilities.new(post: rule)
  
p a.ability
a.ability.call(test: :test)

# => #<Proc:0x007f85c3a2eef0@proc.rb:7>
# => logger: I am logger; hello; {:test=>:test}
```